File: 28.implement-str-str.py
#
# @lc app=leetcode id=28 lang=python3
#
# [28] Implement strStr()
#

# @lc code=start
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        # if len(needle) < 1:
        #     return 0

        # for i in range(len(haystack) - len(needle) + 1):
        #     for j in range(len(needle)):
        #         if needle[j] == haystack[i + j]:
        #             if j == len(needle) - 1:
        #                 # 끝자리까지 다 일치하면
        #                 return i
        #         else:
        #             break

        # return -1

        if needle in haystack:
            return haystack.index(needle)
        else:
            return -1

# @lc code=end

File: Top Interview Questions/src/28.implement-str-str.java
/*
 * @lc app=leetcode id=28 lang=java
 *
 * [28] Implement strStr()
 */

// @lc code=start
class Solution {
  public int strStr(String haystack, String needle) {
    outer: for (int i = 0; i < haystack.length() - needle.length() + 1; i++) {
      for (int j = 0; j < needle.length(); j++) {
        if (haystack.charAt(i + j) == needle.charAt(j)) {
          ;
        } else {
          continue outer;
        }
      }

      return i;
    }

    return -1;
  }
}
// @lc code=end

